src/agent/src/policy.rs

```diff
@@ -200,14 +200,13 @@ impl AgentPolicy {
 
         let results = self.engine.eval_query(query, false)?;
 
-        let prints = match self.engine.take_prints() {
-            Ok(p) => p.join(" "),
-            Err(e) => format!("Failed to get policy log: {e}"),
-        };
-
         if results.result.len() != 1 {
             // Results are empty when AllowRequestsFailingPolicy is used to allow a Request that hasn't been defined in the policy
             if self.allow_failures {
+                let prints = match self.engine.take_prints() {
```

Comment:
I would prefer the single call to take_prints instead of these two here - as in https://github.com/kata-containers/kata-containers/blob/2d9baf899ac16284614a1d3f270ef26e6d1635f6/src/agent/src/policy.rs#L159.

Reply:
Fixed. Thanks!